homeassistant/auth/permissions/util.py

```diff
@@ -1,16 +1,67 @@
 """Helpers to deal with permissions."""
 from functools import wraps
 
-from typing import Callable, Dict, List, Optional, Union, cast  # noqa: F401
+from typing import (  # noqa: F401
+    TYPE_CHECKING, Callable, Dict, List, Optional, Union, cast)
 
+from homeassistant.exceptions import Unauthorized, UnknownUser
+
+from .const import POLICY_CONTROL
 from .models import PermissionLookup
 from .types import CategoryType, SubCategoryDict, ValueType
 
+if TYPE_CHECKING:
+    from homeassistant.core import HomeAssistant, Service, ServiceCall  # noqa
+
 LookupFunc = Callable[[PermissionLookup, SubCategoryDict, str],
                       Optional[ValueType]]
 SubCatLookupType = Dict[str, LookupFunc]
 
 
+def authorized_service_call(hass: 'HomeAssistant', domain: str) -> Callable:
```

Comment:
Should we move it to an auth helper? Traditionally our utils are not aware of `hass`, but helpers are.